provider/src/main/java/com/synopsys/integration/alert/provider/blackduck/issue/BlackDuckProviderIssueHandler.java

```diff
@@ -61,17 +58,14 @@ public void createOrUpdateBlackDuckIssue(String bomComponentVersionIssuesUrl, Bl
         performRequest(requestUri, httpMethod, issueRequestModel);
     }
 
-    private Optional<ProjectVersionIssuesView> retrieveExistingIssue(String projectName, String projectVersionName, String issueKey) throws IntegrationException {
-        Optional<ProjectVersionWrapper> projectVersionWrapperOptional = projectService.getProjectVersion(projectName, projectVersionName);
-        if (projectVersionWrapperOptional.isPresent()) {
-            ProjectVersionWrapper projectVersionWrapper = projectVersionWrapperOptional.get();
-            List<ProjectVersionIssuesView> bomComponentIssues = blackDuckApiClient.getAllResponses(projectVersionWrapper.getProjectVersionView(), ProjectVersionView.ISSUES_LINK_RESPONSE);
-            return bomComponentIssues
-                       .stream()
-                       .filter(issue -> issue.getIssueId().equals(issueKey))
-                       .findAny();
-        }
-        return Optional.empty();
+    private Optional<ProjectVersionIssuesView> retrieveExistingIssue(String projectVersionUrl, String blackDuckIssueId) throws IntegrationException {
+        HttpUrl projectVersionHttpUrl = new HttpUrl(projectVersionUrl);
+        ProjectVersionView projectVersion = blackDuckApiClient.getResponse(projectVersionHttpUrl, ProjectVersionView.class);
+        List<ProjectVersionIssuesView> bomComponentIssues = blackDuckApiClient.getAllResponses(projectVersion, ProjectVersionView.ISSUES_LINK_RESPONSE);
```

Comment:
Is there no servive available that retrieves a list of ProjectVersionIssuesView?

Reply:
Ah, there is now. I don't know if there was one when we originally created the class.